### ISSUE TITLE
Add function method for Fn::Sub

### ISSUE THREAD

**Issue by mwgamble** [mdub/cloud_shaped#6]
There's support for a lot of other CloudFormation functions, but not `Fn::Sub`.

**Comment by mdub** [comment]
That is true enough. I guess I never thought to add direct support for `Fn::Sub`, because I considered the combination of [t.interpolate](https://github.com/mdub/cloud_shaped?tab=readme-ov-file#text-interpolation) and normal old Ruby string interpolation to be sufficient.

Can you say more about what kind of support for `Fn::Sub` you'd want to see, and provide some examples of how it would be useful?
